proto/fulfillment/v1/cluster_type.proto

```diff
@@ -33,9 +33,57 @@ message Cluster {
 }
 
 // The spec contains the details of a cluster as desired by the user.
-//
-// Note that currently this is empty because there are no properties of the cluster that can be modified by the user.
-message ClusterSpec {}
+message ClusterSpec {
+  // Desired node sets of the cluster.
+  //
+  // This will be automatically set by the system when the cluster is initially created, according to the template
+  // selected by the user, and can be later modified to change the size.
+  //
+  // The key of the map is the unique identifier of the node set for this cluster.
+  //
+  // For example, a cluster created with two different node sets, one for nodes without GPUs and another for nodes with
+  // GPUs could be represented like this:
+  //
+  // ```json
+  // {
+  //   "id": "123",
+  //   "spec": {
+  //     "node_sets": {
+  //       "compute": {
+  //         "class": "acme_1tb",
+  //         "size": 3
+  //       },
+  //       "gpu": {
+  //         "class": "acme_1tb_h100",
+  //         "size": 3
+  //       }
+  //     }
+  //   },
+  //   "status": {
+  //     "state": "CLUSTER_STATE_READY",
+  //     "node_sets": {
+  //       "compute": {
+  //         "class": "acme_1tb",
+  //         "size": 3
+  //       },
+  //       "gpu": {
+  //         "class": "acme_1tb_h100",
+  //         "size": 3
+  //       }
+  //     }
+  //   }
+  // }
+  // ```
+  //
+  // The user will be allowed to update the `size` parameter, and the system will decide to allow or reject the request
```

Comment:
By saying that the request will be allowed or rejected, that sounds like the API request itself may be denied. So if I try to go from size 2 to size 10, that could be rejected and the size would remain at 2. But if that was the behavior, it wouldn't really make sense to mark the cluster as degraded.

So what I think you might be describing is that the request will be accepted, and the spec will say 10. But then maybe the system can only get me up to 7 for some reason, and in that case the cluster will be degraded. If this is indeed the behavior you expect, I'd change the description to say something like "if at any time the system cannot allocate the number of nodes requested, the cluster will be marked as degraded with a reason recorded in the condition."

Reply:
I meant the latter, will update the documentation.